### PR TITLE
Remove imports from pip/requests's vendored packages

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Stop relying on vendored packages provided by `pip` and `requests` and instead depend on the dependency itself.

Removes the following fallback imports:
- `pip._vendor.requests`
- `requests.packages.chardet` (added `charset-normalizer` as dependency)
- `pip._vendor.requests.packages.chardet` (added `charset-normalizer` as dependency)

> **Note**
> Opted to depend on `charset-normalizer` instead of `chardet` since the former is also the preferred library used by `requests`.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Review by beeankha — DISMISSED** [review]

File: conda/cli/main_info.py [hunk 1]
```diff
@@ -118,30 +118,14 @@ def print_package_info(packages):
...
+    except ImportError as err:
+        # ImportError: conda-build is not installed
+        log.debug("Unable to import conda-build: %s", err)
         conda_build_version = "not installed"
-    except Exception as e:  # pragma: no cover
-        conda_build_version = "Error %s" % e
-    else:  # pragma: no cover
-        conda_build_version = conda_build.__version__
+    except Exception as err:
+        log.error("Error to import conda-build: %s", err)
```
Comment (kenodegard) [inline conda/cli/main_info.py:127]:
```suggestion
        log.error("Error importing conda-build: %s", err)
```

File: conda/base/context.py [hunk 2]
```diff
@@ -1043,13 +1043,15 @@ def _override(self, key, value):
     @memoizedproperty
     def requests_version(self):
         try:
```
Comment (kenodegard) [inline conda/base/context.py:1045]:
```suggestion
        # used in User-Agent as "requests/<version>"
        # if unable to detect a version we expect "requests/unknown"
        try:
```

**Review by beeankha — DISMISSED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -35,6 +35,7 @@ requirements:
     - tqdm >=4
   run:
     - boltons >=23.0.0
+    - chardet
```
Comment (kenodegard) [inline recipe/meta.yaml:38]:
```suggestion
    - charset-normalizer
```

**Review by beeankha — DISMISSED** [review]

**Review by jezdez — DISMISSED** [review]
charset-normalizer is a pretty old version on defaults: https://anaconda.org/main/charset-normalizer

But a recent one on conda-forge: https://anaconda.org/conda-forge/charset-normalizer

Might be useful to review https://github.com/Ousret/charset_normalizer/blob/master/CHANGELOG.md if there are any features or fixes needed.

Otherwise LGTM

File: conda/cli/main_info.py [hunk 4]
```diff
@@ -118,30 +118,14 @@ def print_package_info(packages):
...
+    except ImportError as err:
+        # ImportError: conda-build is not installed
+        log.debug("Unable to import conda-build: %s", err)
         conda_build_version = "not installed"
-    except Exception as e:  # pragma: no cover
-        conda_build_version = "Error %s" % e
-    else:  # pragma: no cover
-        conda_build_version = conda_build.__version__
+    except Exception as err:
+        log.error("Error importing conda-build: %s", err)
```
Comment (dholth) [inline conda/cli/main_info.py:127]: Could we remove the special ImportError handling and keep except Exception

Comment (kenodegard) [reply conda/cli/main_info.py:127]: of course we can, I'm indifferent either way

Comment (kenodegard) [reply conda/cli/main_info.py:127]: actually they indicate different kinds of information that gets displayed in `conda info` so I think its best to leave this

Comment (dholth) [reply conda/cli/main_info.py:127]: I mean could they be combined to make code coverage testing easier? I see that printing the exception is one way it shows the difference between import and other errors.

**Review by beeankha — APPROVED** [review]

**Pull request merged**